File: Mntone.SplatoonClient/Mntone.SplatoonClient/Entities/Friend.cs
using System;
using System.Diagnostics;
using System.Runtime.Serialization;

namespace Mntone.SplatoonClient.Entities
{
	/// <summary>
	/// Friend information
	/// </summary>
	[DataContract]
	public sealed class Friend
	{
		internal Friend(string id, bool online, string status, Intention intention, string name, Uri miiImageUri)
		{
			this.ID = id;
			this.IsOnline = online;
			this.Status = status;
			this.Intention = intention;
			this.Name = name;
			this.MiiImageUri = miiImageUri;
		}

		/// <summary>
		/// Friend id
		/// </summary>
		[DataMember(Name = "hashed_id")]
		public string ID { get; private set; }

		/// <summary>
		/// Is online
		/// </summary>
		[DataMember(Name = "online")]
		public bool IsOnline { get; private set; }

		/// <summary>
		/// Status
		/// </summary>
		[DataMember(Name = "status")]
		public string Status { get; private set; }

		/// <summary>
		/// Mode
		/// </summary>
		public GameMode Mode { get; private set; }

		[DataMember(Name = "mode")]
		[DebuggerBrowsable(DebuggerBrowsableState.Never)]
		private string ModeImpl
		{
			get { return this.Mode.ToGameModeString(); }
			set { this.Mode = value.ToGameMode(); }
		}

		/// <summary>
		/// Intention
		/// </summary>
		[DataMember(Name = "intention")]
		public Intention Intention { get; private set; }

		/// <summary>
		/// Name
		/// </summary>
		[DataMember(Name = "mii_name")]
		public string Name { get; private set; }

		/// <summary>
		/// Mii iamge uri
		/// </summary>
		[DataMember(Name = "mii_url")]
		public Uri MiiImageUri { get; private set; }
	}
}

File: net.mntone.splatoonclient/net.mntone.splatoonclient.entities/src/main/java/net/mntone/splatoonclient/entities/Friend.java
package net.mntone.splatoonclient.entities;

import org.json.JSONObject;

import java.net.MalformedURLException;
import java.net.URL;

public final class Friend
{
	private final String _id;
	private final boolean _online;
	private final String _status;
	private final GameMode _mode;
	private final Intention _intention;
	private final String _name;
	private final URL _miiImageUri;

	Friend(final String id, final boolean online, final String status, final GameMode mode, final Intention intention, final String name, final URL miiImageUri)
	{
		this._id = id;
		this._online = online;
		this._status = status;
		this._mode = mode;
		this._intention = intention;
		this._name = name;
		this._miiImageUri = miiImageUri;
	}

	Friend(final JSONObject json) throws MalformedURLException
	{
		this._id = json.getString("hashed_id");
		this._online = json.getBoolean("online");
		this._status = json.getString("status");
		this._mode = GameMode.fromText(json.getString("mode"));
		this._intention = new Intention(json.getJSONObject("intention"));
		this._name = json.getString("mii_name");
		this._miiImageUri = new URL(json.getString("mii_url"));
	}

	public String getId()
	{
		return this._id;
	}

	public boolean isOnline()
	{
		return this._online;
	}

	public String getStatus()
	{
		return this._status;
	}

	public GameMode getMode()
	{
		return this._mode;
	}

	public Intention getIntention()
	{
		return this._intention;
	}

	public String getName()
	{
		return this._name;
	}

	public URL getMiiImageUri()
	{
		return this._miiImageUri;
	}
}